src/TaskManagementWeb/Hubs/AgentHub.cs

```diff
@@ -77,8 +77,9 @@ public void TaskFinished(SnTaskResult taskResult)
                     return;
                 }
 
+                bool applicationNeedsNotification = string.IsNullOrWhiteSpace(taskResult.Task.FinalizeUrl) ? false : true;

```

Comment:
I think it is not enough to simply check the task's _finalize URL_ value here, because the ultimate goal is to prevent accessing the client application if it is not necessary. The logic in the **ApplicationHandler.SendFinalizeNotificationAsync** method uses **task.GetFinalizeUrl** to get the 'final' finalize URL - which can be the _global_ finalize url, if the task does not have one. 

So I would use the same method to determine whether a ping should be sent or not.